filters/openpolicyagent/opaauthorizerequest/opaauthorizerequest.go

```diff
@@ -143,8 +152,33 @@ func addRequestHeaders(fc filters.FilterContext, headers http.Header) {
 		}
 	}
 }
+func (f *opaAuthorizeRequestFilter) Response(fc filters.FilterContext) {
+	span, _ := f.opa.StartSpanFromFilterContext(fc)
+	defer span.Finish()
+
+	headersToAdd := fc.StateBag()[openpolicyagent.ResponseHeadersKey]
+	if headersToAdd == nil {
+		return
+	}
 
-func (*opaAuthorizeRequestFilter) Response(filters.FilterContext) {}
+	var headers []*ext_authz_v3_core.HeaderValueOption
+	if headerValues, ok := headersToAdd.([]*ext_authz_v3_core.HeaderValueOption); ok {
+		headers = headerValues
+	} else {
+		f.opa.HandleInvalidDecisionError(fc, span, nil, fmt.Errorf("failed to read response headers: %v", headersToAdd), !f.opa.EnvoyPluginConfig().DryRun)
+		return
+	}
```

Comment:
Just check that statebag value type is *ext_core_v3.HeaderValueOption - we do not need complicated error handling here as we put the value into statebag ourselves in the Request.
```go
if headers, ok := ctx.StateBag()[openpolicyagent.ResponseHeadersKey].(*[]*ext_authz_v3_core.HeaderValueOption); ok {
   addResponseHeaders(fc, headers)
}
```

Reply:
Thanks a lot for the snippet. Addressed.